File: acoustics_toolbox_r2020/Python/Bellhop/Tests/SW/Calib/calibB.py
#==================================================================
#  
#  Bellhop: Calibration case
#  Mexilhoeira Grande, Dom Jul 30 21:14:17 WEST 2017
#  Written by Tordar
#  
#==================================================================

# ipython: run calibB

from os import *
import sys
from numpy import *
from scipy.io import *
from pylab import *
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from readshd import *

rs = 0.0

print("Bellhop - Calibration case:") 
print("Coherent TL calculation")
print("Geometric hat beams in Cartesian coordinates")

system("bellhop.exe calibB")

filename = 'calibB.shd'
xs = nan
ys = nan
pressure,geometry = readshd(filename,xs,ys)

zs     = geometry["zs"]
rarray = geometry["rarray"]; rarraykm = rarray/1000
zarray = geometry["zarray"]

Dmax = zarray[-1]
rmax = rarray[-1]; rmaxkm = rmax/1000

p = squeeze( pressure, axis=(0,1) )
tl = -20*log10( abs( p ) )

figure(1)
imshow(tl,extent=[0,rmaxkm,0,Dmax],aspect='auto',cmap='jet_r',origin='lower')
cb = colorbar()
cb.ax.invert_yaxis()
plot(rs,zs,marker="<",markersize=16,color="k")
xlabel('Range (km)')
ylabel('Depth (m)')
title('Bellhop - Calibration case')
ylim(Dmax,0)

show()

print("done.")

File: acoustics_toolbox_r2020/Python/Bellhop/Tests/SW/Calib/calibB.m
clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite

disp('Bellhop - calibration:')
disp('TL calculation') 

system('bellhop.exe calibB');

[ PlotTitle, PlotType, freq, atten, Pos, p ] = read_shd( 'calibB.shd' );

zs = Pos.s.depth;

rarray = Pos.r.range; rarraykm = rarray/1000;
zarray = Pos.r.depth;

rmax = max( rarray ); rmaxkm = rmax/1000;
zmax = max( zarray );

pressure = squeeze( p );

tl = -20*log10( abs( pressure ) ); 

tej = flipud( jet( 256 ) );

figure(1), hold on
plot(0,zs,'ko',0,zs,'m*','MarkerSize',16) 
pcolor(rarraykm,zarray,tl), shading interp, colormap( tej ), caxis([40 90]), 
h = colorbar;
set( h, 'YDir', 'reverse' );
axis([0 rmaxkm 0 zmax])
hold off
view(0,-90)
xlabel('Range (km)')
ylabel('Depth (m)')
title('Bellhop - calibration , 12 kHz')

disp('done.')
